dipy/viz/widget.py

```diff
@@ -272,40 +283,41 @@ def text(iren, ren, callback, message="DIPY",
     class TextWidget(vtk.vtkTextWidget):
 
         def place(self, renderer):
-
             text_rep = self.GetRepresentation()
 
             position = text_rep.GetPositionCoordinate()
-            position.SetCoordinateSystemToNormalizedDisplay()
-            position.SetValue(*left_down_pos)
-
             position2 = text_rep.GetPosition2Coordinate()
-            position2.SetCoordinateSystemToNormalizedDisplay()
-            position2.SetValue(*right_top_pos)
 
-            self.SelectableOn()
-            self.ResizableOff()
-            text_rep.ProportionalResizeOn()
+            # The dtype of `left_down_pos` determines coordinate system type.
+            if np.issubdtype(np.asarray(left_down_pos).dtype, np.integer):
+                position.SetCoordinateSystemToDisplay()
+            else:
+                position.SetCoordinateSystemToNormalizedDisplay()
 
-            self.On()
+            # The dtype of `right_top_pos` determines coordinate system type.
+            if np.issubdtype(np.asarray(right_top_pos).dtype, np.integer):
+                position2.SetCoordinateSystemToDisplay()
+            else:
+                position2.SetCoordinateSystemToNormalizedDisplay()
+
+            position.SetValue(*left_down_pos)
+            position2.SetValue(*right_top_pos)
 
     text_widget = TextWidget()
     text_widget.SetRepresentation(text_rep)
     text_widget.SetInteractor(iren)
-    text_widget.SetTextActor(text_actor)
     text_widget.SelectableOn()
     text_widget.ResizableOff()
-    text_widget.GetRepresentation().ProportionalResizeOn()
 
     text_widget.AddObserver(vtk.vtkCommand.WidgetActivateEvent, callback)
-    text_widget.On()
 
-    # This is a hack for avoiding not plotting the text widget when
-    # backface culling in On on a different actor
-    ss = vtk.vtkSphereSource()
-    mapper = vtk.vtkPolyDataMapper()
-    mapper.SetInputConnection(ss.GetOutputPort())
-    actor = vtk.vtkActor()
-    actor.GetProperty().BackfaceCullingOff()
```

Comment:
With VTK6.0, on my machine, unless the mapper is added to the actor and the actor is added to the renderer, there is no difference.